### PR TITLE
[ARTS-325] JsonSchema generator tool 

### PR DESCRIPTION
**Pull request by anatbal**
**Updated PR, with the right format**
## Description
Java code of jsonSchema generator, including a jar file and a README file that explains the use of the tool.

### Changes
- [ARTS-325]- Build data entities for the json schema- all relevant models are under model folder, perhaps in the future we will want to create a consistency between the trial config models and the models that are necessary today for the terraform generation.

### Checklist:

- [ ] Branch name follows convention (feature/arts-###-short-name OR fix/arts-###-short-name)
- [ ] I have included a short-meaningful title, description and changes for this PR

[ARTS-325]: https://ndph-arts.atlassian.net/browse/ARTS-325

### CODE REVIEW

File: build-tools/src/main/java/uk/ac/ox/ndph/mts/json_schema_generator/model/TrialConfigJsonSchema.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    private Service siteService;
+
+    @JsonProperty(value = "practitioner_service", required = true)
+    private Service practitionerService;
+
+    @JsonProperty(value = "config_server_service", required = true)
+    private Service configServerService;
+
+    @JsonProperty(value = "spring_cloud", required = true)
+    private SpringCloud springCloud;
```
Comment (tamirkamara) [inline build-tools/src/main/java/uk/ac/ox/ndph/mts/json_schema_generator/model/TrialConfigJsonSchema.java:34]: Can we rename this to SpringCloudServices?
I think it's more self explanatory

**Review by tamirkamara — CHANGES_REQUESTED** [review]
Looks nice, added a few comments.

File: build-tools/src/main/java/uk/ac/ox/ndph/mts/json_schema_generator/model/TrialConfigJsonSchema.java (hunk 1 above)
Comment (anatbal) [reply build-tools/src/main/java/uk/ac/ox/ndph/mts/json_schema_generator/model/TrialConfigJsonSchema.java]: I agree, changed.

File: build-tools/src/main/java/uk/ac/ox/ndph/mts/json_schema_generator/Main.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+
+public final class Main {
+    static String fileName = "jsonSchema.json";
+
+    public static void main(final String[] args) throws IOException {
+        SchemaGenerator jsonSchemaGenerator;
+
+        // Generate JsonSchema object
+        jsonSchemaGenerator = new SchemaGenerator(TrialConfigJsonSchema.class);
```
[the remark this answers is not in the document]
Comment (anatbal) [reply build-tools/src/main/java/uk/ac/ox/ndph/mts/json_schema_generator/Main.java]: Yes, got a little confused there, now I see, fixed

File: build-tools/src/main/java/uk/ac/ox/ndph/mts/json_schema_generator/Main.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        JsonNode jsonSchema = jsonSchemaGenerator.getJsonSchema();
+
+        // Write JsonSchema to file
+        ObjectMapper jsonObjectMapper = new ObjectMapper();
+
+        // Apply indentation
+        jsonObjectMapper.enable(SerializationFeature.INDENT_OUTPUT);
+        jsonObjectMapper.writeValue(new File(fileName), jsonSchema);
```
[the remark this answers is not in the document]
Comment (anatbal) [reply build-tools/src/main/java/uk/ac/ox/ndph/mts/json_schema_generator/Main.java:27]: Changed

File: build-tools/pom.xml [hunk 4]
```diff
@@ -7,5 +7,46 @@
...
     <version>1.0</version>
-
-</project>
\ No newline at end of file
+    <parent>
+        <groupId>uk.ac.ox.ndph.mts</groupId>
+        <artifactId>mts-services-parent</artifactId>
+        <version>0.0.1-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
```
[the remark this answers is not in the document]
Comment (anatbal) [reply build-tools/pom.xml]: We probably don't need the whole parent, but I had to add at least the 'maven-compiler-plugin' in the pom.xml in order to compile the project.

**Review by tamirkamara — APPROVED** [review]

**Review by eladiw — APPROVED** [review]

**Pull request merged**